### ISSUE TITLE
Web GUI freezes if you leave it alone on Firefox on GNU/Linux.

### ISSUE THREAD

**Issue by veronikazaglotova** [pixeltris/GK6X#99]
I used the GUI. Everything was ok. Then I switched to another window for a few seconds. Then I do all the changes I want to do, click "Save" "Apply"... NOTHING happens. The apply button has a loading animation on it. I restart the gui server from my terminal and everything works fine, until I switch to another window for a few seconds and it stops working again

**Comment by pixeltris** [comment]
The web GUI hasn't been fully tested. When the apply button fails there's usually errors in the web browser console which might give an indication of what's going wrong (certain API end points failing, etc).

**Comment by veronikazaglotova** [comment]
If I understood correctly, the GUI uses a client-server system (the GK6X.exe starts a server, then I connect to it from the browser). When the apply button fails, I try to restart the page and Firefox says "Unable to connect: timeout" or something. I think the server freezes or something wrong going with it.
I'll send the Web console output later.

**Comment by pixeltris** [comment]
Yes that's correct. There is a ping timeout in the dedicated UI mode which will close the server if the tab is closed, but I don't think that's even set up at the moment https://github.com/pixeltris/GK6X/blob/470bd0733909b41ffedf6b5334126f84dbfda479/GK6X/Program.cs#L134

**Comment by veronikazaglotova** [comment]
I didn't close the tab, I just switched for a sec :/
Why not make this configurable?

**Comment by pixeltris** [comment]
That code path should only run in dedicated UI mode. The dedicated UI mode is something that isn't complete/bundled yet. It's designed to be separate thing from the CLI, which runs as a background process. To ensure it closes after the web-GUI is closed it pings any open tabs. After a period of time of no interaction the background process kills itself.
